climakitae/explore/typical_meteorological_year.py

```diff
@@ -0,0 +1,804 @@
+        if self.weighted_fs_sum is UNSET:
+            self.set_weighted_statistic()
+        self._vprint("Finding top months (lowest F-S statistic)")
+        self.top_months = get_top_months(self.weighted_fs_sum)
+        return
+
+    def show_tmy_data_to_export(self, simulation: str):
+        """Show line plots of TMY data for single model.
+
+        Parameters
+        ----------
+        simulation: str
+            Simulation to display.
+
+        """
+        if self.tmy_data_to_export is UNSET:
+            print("No TMY data generated.")
+            print("Please run TMY.generate_tmy() to create TMY data for viewing.")
+            return
+
+        self.tmy_data_to_export[simulation].plot(
+            x="time",
+            y=list(self.vars_and_units.keys()),
+            title=f"Typical Meteorological Year ({simulation})",
+            subplots=True,
+            figsize=(10, 8),
+            legend=True,
+        )
+        return
+
+    def run_tmy_analysis(self):
+        """Generate typical meteorological year data
+        Output will be a list of dataframes per simulation.
+        Print statements throughout the function indicate to the user the progress of the computatioconvert_to_local_time
+
+        Parameters
+        -----------
+        top_df: pd.DataFrame
+            Table with column values month, simulation, and year
+            Each month-sim-yr combo represents the top candidate that has the lowest weighted sum from the FS statistic
+
+        Returns
+        --------
+        dict of str: pd.DataFrame
+            Dictionary in the format of {simulation:TMY corresponding to that simulation}
+
+        """
+        print("Assembling TMY data to export. Expected runtime: 30 minutes")
+
+        self._vprint("  STEP 1: Retrieving hourly data from catalog")
+        # Loop through each variable and grab data from catalog
+        all_vars_list = []
+
+        for var, units in self.vars_and_units.items():
+            print(f"  Getting {var}", end="... ")
+            data_by_var = self._load_single_variable(var, units)
+
+            # Drop unwanted coords
+            data_by_var = data_by_var.squeeze().drop_vars(
+                ["lakemask", "landmask", "x", "y", "Lambert_Conformal"]
+            )
+
+            all_vars_list.append(data_by_var)  # Append to list
+
+        # Merge data from all variables into a single xr.Dataset object
+        all_vars_ds = xr.merge(all_vars_list)
+
+        # Construct TMY
+        self._vprint(
+            "\n  STEP 2: Calculating Typical Meteorological Year per model simulation\nProgress bar shows code looping through each month in the year.\n"
+        )
+
+        self.tmy_data_to_export = self._make_8760_tables(
+            all_vars_ds, self.top_months
+        )  # Return dict of TMY by simulation
+        self._vprint("TMY analysis complete")
+        return
+
+    def export_tmy_data(self, extension: str = "tmy"):
```

Comment:
I'd recommend for now having the default extension be "epw". Eventually, we'll add in "csv" (and have it be the default).